docs/_integration-with-existing-apps-ios.md

```diff
@@ -0,0 +1,596 @@
++#else
++    Bundle.main.url(forResource: "main", withExtension: "jsbundle")
++#endif
++  }
+}
+```
+
+Let's have a look at the code above:
+
+1. We are inherithing from the `RCTAppDelegate` and we are calling the `applicatinon(_:didFinishLaunchingWithOptions:)` of the `RCTAppDelegate`. This delegates all the React Native initialization processes to the base class.
+2. We are customizing the `RCTAppDelegate` by setting the `automaticallyLoadReactNativeWindow` to `false`. This step instruct React Native that the app is handling the `UIWindow` and React Native should not worry about that.
+3. The methods `sourceURLForBridge(for:)` and `bundleURL()` are used by the App to tell to React Native wher it can find the JS bundle that needs to be rendered. The `sourceURLForBridge(for:)` is from the Old Architecture and you can see that it is deferring the decision to the `bundleURL()` method, required by the New Architecture.
+
+</TabItem>
+</Tabs>
+
+#### Presenting a React Native view in a rootViewController
+
+Finally, we can present our React Native view. To do so, we need a new View Controller that can host a view in which we can load the JS content.
+
+1. From Xcode, let's create a new `UIViewController` (Let's call it `ReactViewController`).
+2. Have the Initial `ViewController` present the `ReactViewController`. There are several ways to do so, depending on your app. For this example, we assume that you have a button that presents React Native Modally.
+
+<Tabs groupId="ios-language" queryString defaultValue={constants.defaultAppleLanguage} values={constants.appleLanguages}>
+<TabItem value="objc">
+
+```diff title="ViewController.m"
+#import "ViewController.h"
++#import "ReactViewController.h"
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+ - (void)viewDidLoad {
+   [super viewDidLoad];
+   // Do any additional setup after loading the view.
+   self.view.backgroundColor = UIColor.systemBackgroundColor;
++  UIButton * button = [UIButton new];
++  [button setTitle:@"Open React Native" forState:UIControlStateNormal];
++  [button setTitleColor:UIColor.systemBlueColor forState:UIControlStateNormal];
++  [button setTitleColor:UIColor.blueColor forState:UIControlStateHighlighted];
++  [button addTarget:self action:@selector(presentReactNative) forControlEvents:UIControlEventTouchUpInside];
++  [self.view addSubview:button];
+
++  button.translatesAutoresizingMaskIntoConstraints = NO;
++  [NSLayoutConstraint activateConstraints:@[
++    [button.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor],
++    [button.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor],
++    [button.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor],
++    [button.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
++  ]];
+ }
+
++- (void)presentReactNative
++{
++  [self presentViewController:[ReactViewController new] animated:YES completion:nil];
```

Comment:
Shouldn't we retain this ViewController as a property on the class?

I am not sure but I think I've run into some memory management issues with deallocation when doing it this way

Reply:
It depends on what the VC does. In this simple example it should not be required, but let add it for safety.